var/spack/repos/builtin/packages/openmpi/package.py

```diff
@@ -41,12 +41,14 @@ class Openmpi(Package):
     def url_for_version(self, version):
         return "http://www.open-mpi.org/software/ompi/v%s/downloads/openmpi-%s.tar.bz2" % (version.up_to(2), version)
 
-    def setup_dependent_environment(self, module, spec, dep_spec):
-        """For dependencies, make mpicc's use spack wrapper."""
-        os.environ['OMPI_CC'] = 'cc'
-        os.environ['OMPI_CXX'] = 'c++'
-        os.environ['OMPI_FC'] = 'f90'
-        os.environ['OMPI_F77'] = 'f77'
+    def environment_modifications(self, dependent_spec):
+        env = super(Openmpi, self).environment_modifications(dependent_spec)
+        # FIXME : the compilers should point to the current wrappers, not to generic cc etc.
```

Comment:
@tgamblin I think this is solved by the splitting of `setup_environment` and `setup_dependent_environment`, see `openmpi` or `mpich` packages in this PR for an example of use.